tests/neg/i11694.scala

```diff
@@ -0,0 +1,19 @@
+def test1 = {
+  def f11: (Int => Int) | Unit = x => x + 1
+  def f12: Null | (Int => Int) = x => x + 1
```

Comment:
Shouldn't this be 
```scala
def f12: Unit | (Int => Int) = x => x + 1
```

otherwise f12 and f22 are identical.

Reply:
Yes, you are right😂